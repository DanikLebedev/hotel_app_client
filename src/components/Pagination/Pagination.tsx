import React from 'react';
import { func, number } from 'prop-types';

interface Pagination {
    postPerPage: number;
    totalPosts: number;
    paginate: (pageNumber: number) => void;
    currentPage: number;
}

export const Pagination: React.FC<Pagination> = ({ postPerPage, totalPosts, paginate, currentPage }) => {
    const pageNumbers: number[] = [];

    for (let i = 1; i <= Math.ceil(totalPosts / postPerPage); i++) {
        pageNumbers.push(i);
    }

    return (
        <>
            <ul className="pagination">
                {pageNumbers.map(number => (
                    <li key={number} className={currentPage === number ? 'page-item active' : 'page-item'}>
                        <p onClick={() => paginate(number)} className="page-link">
                            {number}
                        </p>
                    </li>
                ))}
            </ul>
        </>
    );
};

Pagination.propTypes = {
    postPerPage: number.isRequired,
    totalPosts: number.isRequired,
    paginate: func.isRequired,
    currentPage: number.isRequired,
};
