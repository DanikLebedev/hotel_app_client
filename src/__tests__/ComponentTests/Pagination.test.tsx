import { mount, shallow } from 'enzyme';
import React from 'react';
import { Pagination } from '../../components/Pagination/Pagination';

it('should render without crashing', function() {
    const paginate = jest.fn();
    const wrapper = shallow(<Pagination currentPage={2} paginate={paginate} postPerPage={2} totalPosts={10} />);
    expect(wrapper).toMatchSnapshot();
});

it('should change active item ', function() {
    const paginate = jest.fn();
    const wrapper = mount(<Pagination currentPage={2} paginate={paginate} postPerPage={2} totalPosts={10} />);
    expect(wrapper.find('.page-item .active-link')).toBeDefined();
});
