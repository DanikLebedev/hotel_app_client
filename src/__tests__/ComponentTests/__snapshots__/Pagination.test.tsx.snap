// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<Fragment>
  <ul
    className="pagination"
  >
    <li
      className="page-item"
      key="1"
    >
      <p
        className="page-link"
        color="#000"
        onClick={[Function]}
      >
        1
      </p>
    </li>
    <li
      className="page-item active-link"
      key="2"
    >
      <p
        className="page-link"
        color="#000"
        onClick={[Function]}
      >
        2
      </p>
    </li>
    <li
      className="page-item"
      key="3"
    >
      <p
        className="page-link"
        color="#000"
        onClick={[Function]}
      >
        3
      </p>
    </li>
    <li
      className="page-item"
      key="4"
    >
      <p
        className="page-link"
        color="#000"
        onClick={[Function]}
      >
        4
      </p>
    </li>
    <li
      className="page-item"
      key="5"
    >
      <p
        className="page-link"
        color="#000"
        onClick={[Function]}
      >
        5
      </p>
    </li>
  </ul>
</Fragment>
`;
